src/main/java/org/jboss/fuse/forge/addon/ui/FuseProjectSetupStep.java

```diff
@@ -0,0 +1,273 @@
+        return Metadata.from(super.getMetadata(context), getClass())
+            .name("JBoss Fuse: Choose Archetype")
+            .description("Choose a new JBoss Fuse archetype to generate your project");
+    }
+
+    @Override
+    public void initializeUI(UIBuilder builder) throws Exception {
+        Coordinate coordinate = MavenUtils.createCoordinate(ARCHETYPE_CATALOG_GROUP_ID, ARCHETYPE_CATALOG_ARTIFACT_ID);
+        archetypeVersions = MavenUtils.resolveVersions(resolver, coordinate);
+
+        configureProjectTypeInput();
+        configureCatalogVersionInput();
+        configureArchetypeInput();
+
+        builder.add(fuseProjectType).add(catalogVersion).add(archetype);
+    }
+
+    @Override
+    public NavigationResult next(UINavigationContext context) throws Exception {
+        return null;
+    }
+
+    @Override
+    public Result execute(UIExecutionContext context) throws Exception {
+        Archetype selectedArchetype = archetype.getValue();
+        String repository = selectedArchetype.getRepository();
+        String coordinate = String.format("%s:%s:%s", selectedArchetype.getGroupId(), selectedArchetype.getArtifactId(),
+            selectedArchetype.getVersion());
+        DependencyQueryBuilder depQuery = DependencyQueryBuilder.create(coordinate);
+
+        if (!Strings.isNullOrEmpty(repository)) {
+            if (repository.endsWith(".xml")) {
+                int lastRepositoryPath = repository.lastIndexOf('/');
+                if (lastRepositoryPath > -1) {
+                    repository = repository.substring(0, lastRepositoryPath);
+                }
+            }
+            if (!repository.isEmpty()) {
+                depQuery.setRepositories(new DependencyRepository("archetype", repository));
+            }
+        }
+
+        Dependency resolvedArtifact = resolver.resolveArtifact(depQuery);
+        FileResource<?> artifact = resolvedArtifact.getArtifact();
+
+        UIContext uiContext = context.getUIContext();
+        Project project = (Project) uiContext.getAttributeMap().get(Project.class);
+        MetadataFacet metadataFacet = project.getFacet(MetadataFacet.class);
+        File fileRoot = project.getRoot().reify(DirectoryResource.class).getUnderlyingResourceObject();
+
+        ArchetypeHelper archetypeHelper = new ArchetypeHelper(artifact.getResourceInputStream(), fileRoot, metadataFacet.getProjectGroupName(),
+            metadataFacet.getProjectName(), metadataFacet.getProjectVersion());
+
+        JavaSourceFacet facet = project.getFacet(JavaSourceFacet.class);
+        archetypeHelper.setPackageName(facet.getBasePackage());
+        archetypeHelper.execute();
+
+        return Results.success("Created new JBoss Fuse project in: " + fileRoot.getPath());
+    }
+
+    private void configureProjectTypeInput() {
+        Set<String> types = new LinkedHashSet<>();
+        for (FuseProjectCategory category : FuseProjectCategory.values()) {
```

Comment:
You can also use map on a stream, you will have somethign like:
Set<String> types = FuseProjectCategory.values().stream()
.map(category -> category.getName())
.collect(Collectors.toCollection(LinkedHashSet::new));